Jint/Runtime/TypeConverter.cs

```diff
@@ -845,84 +858,156 @@ public static void CheckObjectCoercible(Engine engine, JsValue o)
-                var score = 0;
-                var parameters = tuple.Item1.Parameters;
-                var arguments = tuple.Item2;
-                for (var i = 0; i < arguments.Length; i++)
-                {
-                    var jsValue = arguments[i];
-                    var arg = jsValue.ToObject();
-                    var argType = arg?.GetType();
-                    var paramType = parameters[i].ParameterType;
-                    if (arg == null)
-                    {
-                        if (!TypeIsNullable(paramType))
-                        {
-                            score -= 10000;
-                        }
-                    }
-                    else if (paramType == typeof(JsValue))
-                    {
-                        // JsValue is convertible to. But it is still not a perfect match
-                        score -= 1;
-                    }
-                    else if (argType != paramType)
-                    {
-                        // check if we can do conversion from int value to enum
-                        if (paramType.IsEnum &&
-                            jsValue is JsNumber jsNumber
-                            && jsNumber.IsInteger()
-                            && Enum.IsDefined(paramType, jsNumber.AsInteger()))
-                        {
-                            // OK
-                        }
-                        else
-                        {
-                            if (paramType.IsAssignableFrom(argType))
-                            {
-                                score -= 10;
-                            }
-                            else
-                            {
-                                if (argType.GetOperatorOverloadMethods()
-                                    .Any(m => paramType.IsAssignableFrom(m.ReturnType) &&
-                                              (m.Name == "op_Implicit" ||
-                                               m.Name == "op_Explicit")))
-                                {
-                                    score -= 100;
-                                }
-                                else
-                                {
-                                    score -= 1000;
-                                }
-                            }
-                        }
-                    }
-                }
+                yield return match;
+            }
+        }
+
+        /// <summary>
+        /// Method's match score tells how far away it's from ideal candidate. 0 = ideal, bigger the the number,
+        /// the farther away the candidate is from ideal match. Negative signals impossible match.
+        /// </summary>
+        private static int CalculateMethodScore(Engine engine, MethodDescriptor method, JsValue[] arguments)
+        {
+            if (method.Parameters.Length == 0 && arguments.Length == 0)
+            {
+                // perfect
+                return 0;
+            }
```

Comment:
Does this mean method will match perfectly if argument lengths are the same, regardless of the argument types? Isn't this wrong, or am I missing something?

Reply:
Method has no parameters and JS call has no arguments? What is there to match?

Reply:
I had a little brain fart. I saw another thing when reviewing.

Sorry for the trouble. Thank you.

Reply:
No worries, I always appreciate reviews and and extra pair of eyes 👍🏻